### PR TITLE
drop py2 support + convert README to md + travis improvements

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by timj** [comment]
There are some `builtins` imports that also need to be removed.

**Comment by timj** [comment]
There are also some flake8 warnings (at least on master):
```
$ flake8
./python/lsst/ci/build.py:190:38: W605 invalid escape sequence '\.'
./python/lsst/ci/prepare.py:116:31: W605 invalid escape sequence '\w'
./python/lsst/ci/prepare.py:644:17: E117 over-indented (comment)
./python/lsst/ci/prepare.py:645:17: E117 over-indented
./python/lsst/ci/prepare.py:646:25: E117 over-indented
./python/lsst/ci/prepare.py:648:33: E117 over-indented
```

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +12,6 @@ matrix:
         python setup.py test
         # https://github.com/tholo/pytest-flake8/issues/27
         flake8 bin/lsst-build
```
Comment (timj) [inline .travis.yml:31]: Why isn't this checking the other python code?

Comment (jhoblitt) [reply .travis.yml:31]: see the comment above -- I couldn't figure out how to get it to check one file without a `.py` extension.

Comment (timj) [reply .travis.yml:31]: Which is fine, but I think from what I see there that we don't then test any of the other python code. Adding a second call to flake8 is needed I think (or adding `python/` to the arguments for that flake8 call).

Comment (jhoblitt) [reply .travis.yml:31]: All of the python code is flake8'd by `pytest-flake8`, only the bin is being missed and that should go away if it's converted to a console script.

**Comment by jhoblitt** [comment]
I've removed the usage of `buildtins` but something with `pytest-flake8` is now broken -- the versions in `setup.py` probably need to be reviewed but I've got to run.

File: .travis.yml (hunk 1 above)
Comment (timj) [reply .travis.yml:31]: How do you explain the flake8 problems on master then?

**Comment by timj** [comment]
@jhoblitt you need pytest-flake8 1.0.4 to work with the flake8 >= 3.7.5 (which is the version we use elsewhere).

File: .travis.yml (hunk 1 above)
Comment (jhoblitt) [reply .travis.yml:31]: flake8 wasn't complaining *a year ago* when master was last built.  flake8 output is visible in the console of the most recent build: https://travis-ci.org/lsst/lsst_build/jobs/339213143

**Comment by jhoblitt** [comment]
@timj This PR should be ready for review now.

File: .mdlrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# a seperate "style" file must be used to pass "parameters" to a rule
```
Comment (timj) [inline .mdlrc]: seperate -> separate

File: README.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    build lsst_distrib
+
+lsst-build build build
+```
+
+`lsst-build prepare`
+------------------
+
+`lsst-build prepare` clones the given top level product(s) into `<builddir>`,
+discovers its depenencies by reading its `ups/$PRODUCT.table` file, and
```
Comment (timj) [inline README.md]: depenencies -> dependencies

**Review by timj — APPROVED** [review]
Looks good to me. At some point we should consider moving much of the setup.py configuration parameters into the setup.cfg (setuptools has supported that for a long time). Thanks for the great documentation although I'm surprised it's not in reST format. Would @jonathansick prefer it in sphinx form?

File: .mdlrc (hunk 2 above)
Comment (jhoblitt) [reply .mdlrc]: fixed.

File: README.md (hunk 3 above)
Comment (jhoblitt) [reply README.md]: fixed.

**Comment by jhoblitt** [comment]
@timj I'm not the originator of the markdown -- it was the original format in the `README`. What I did was fix up some problems and get it to pass `markdownlint`.  I have no objection to it being converting to rst.  The main objective of this PR was to get the "tests" back into a working state.

**Pull request merged**